lib/rspec/sidekiq/matchers/be_deadable.rb

```diff
@@ -0,0 +1,41 @@
+module RSpec
+  module Sidekiq
+    module Matchers
+      def be_deadable
+        BeDeadable.new(false)
+      end
+
+       def not_be_deadable
+        BeDeadable.new(true)
+      end
```

Comment:
Sorry for the confusion, I didn't understand some part of your first comments. Now is so much clear for me. Thank you for your feedback!
This will be my first open source contribution 😃 .
Today/tomorrow I will  make changes based into your CR.
Cheers